pkg/orchestration/wiring/saml/plugin.go

```diff
@@ -0,0 +1,122 @@
+	ResourcePrefix                                     = "SAML"
+)
+
+type autowiringOnlySpec struct {
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	URL         string `json:"url"`
+}
+type finalSpec struct {
+	autowiringOnlySpec
+}
+type WiringPlugin struct {
+}
+
+func New() *WiringPlugin {
+	return &WiringPlugin{}
+}
+
+func (p *WiringPlugin) WireUp(resource *orch_v1.StateResource, context *wiringplugin.WiringContext) wiringplugin.WiringResult {
+	if resource.Type != ResourceType {
+		return &wiringplugin.WiringResultFailure{
+			Error: errors.Errorf("invalid resource type: %q", resource.Type),
+		}
+	}
+
+	serviceInstance, err := osb.ConstructServiceInstance(resource, clusterServiceClassExternalID, clusterServicePlanExternalID)
+	if err != nil {
+		return &wiringplugin.WiringResultFailure{
+			Error: err,
+		}
+	}
+
+	instanceParameters, external, retriable, err := instanceParameters(resource, context)
+	if err != nil {
+		return &wiringplugin.WiringResultFailure{
+			Error:            err,
+			IsExternalError:  external,
+			IsRetriableError: retriable,
+		}
+	}
+	serviceInstance.Spec.Parameters = &runtime.RawExtension{
+		Raw: instanceParameters,
+	}
+
+	instanceResourceName := wiringutil.ServiceInstanceResourceName(resource.Name)
+
+	smithResource := smith_v1.Resource{
+		Name:       instanceResourceName,
+		References: nil,
+		Spec: smith_v1.ResourceSpec{
+			Object: serviceInstance,
+		},
+	}
+
+	return &wiringplugin.WiringResultSuccess{
+		Contract: wiringplugin.ResourceContract{
+			Shapes: []wiringplugin.Shape{
+				knownshapes.NewBindableEnvironmentVariablesWithExcludeResourceName(smithResource.Name, ResourcePrefix, map[string]string{
+					"SAML_IDP_METADATA_URL": "data.saml_idp_metadata_url",
+					"IDP_METADATA_URL":      "data.idp_metadata_url",
+				}, true),
+			},
+		},
+		Resources: []smith_v1.Resource{smithResource},
+	}
+}
+
+func instanceParameters(resource *orch_v1.StateResource, context *wiringplugin.WiringContext) ([]byte, bool /* externalErr */, bool /* retriable */, error) {
+	if len(context.Dependencies) > 0 {
+		return nil, true, false, errors.New("saml resource should not have any dependencies")
+	}
+	if resource.Spec == nil {
+		return nil, true, false, errors.New("saml resource is missing user parameters")
+	}
+
+	var spec finalSpec
+
+	err := json.Unmarshal(resource.Spec.Raw, &spec)
+	if err != nil {
+		return nil, false, false, errors.WithStack(err)
+	}
+
+	// Set default name service--location--resource
+	if spec.Name == "" {
+		label := ""
+		if context.StateContext.Location.Label != "" {
+			label = fmt.Sprintf(".%s", context.StateContext.Location.Label)
+		}
+		spec.Name = fmt.Sprintf("%s--%s.%s.%s%s--%s",
+			context.StateContext.ServiceName,
+			context.StateContext.Location.EnvType,
+			context.StateContext.Location.Account,
+			context.StateContext.Location.Region,
+			label,
+			resource.Name,
+		)
+	}
+
+	out, err := json.Marshal(&spec)
```

Comment:
with this approach only parameters defined in the struct will be passed through OSB, is this intended? the code for some other OSB brokers, e.g. OAP, allows for passing through "unknown" parameters that are not defined in the struct.

to me this is fine, especially given the recent agreement to use camel case for all parameters, and it seems that SAM provider is using snake casing (based on response env vars naming). just wanted to double check.